### PR TITLE
Fix broken migration

### PR DESCRIPTION
**Pull request by cammellos**
The migration from unencrypted->encrypted did not work properly, it was
deleting and re-creating the database. This correctly migrates the db.

I really thought I tested it and was fine, but I must have missed something.

Please hold on merging until, I want to run some tests on the build.

### CODE REVIEW

**Review by mandrigin — APPROVED** [review]
lgtm

**Review by rasom — APPROVED** [review]

**Review by dshulyak — APPROVED** [review]

**Review by mandrigin — CHANGES_REQUESTED** [review]
Hmm, too many commits?

**Comment by cammellos** [comment]
I am building on top of pedro's in order to create a build for testing (otherwise is not compatible with status-react). I will remove them before merging :)

**Comment by mandrigin** [comment]
So, I will keep my review until that, so the PR isn’t accidentally merged.

**Review by adambabik — CHANGES_REQUESTED** [review]
Please rebase also. I see some changes in `account/` which are already merged https://github.com/status-im/status-go/pull/1346.

**Comment by cammellos** [comment]
@adambabik @mandrigin this is now ready to be merged. 
Could you remove the request for changes?

@adambabik  It has been already tested by the test-team for that particular issue, so if you could create a release with the whole develop after merged please.

I will include this release to https://github.com/status-im/status-react/pull/7294 including the changes in the config to make it compatible  https://github.com/status-im/status-react/pull/7227 , there's already a build that had been tested with these changes by the test team.

**Review by mandrigin — APPROVED** [review]

File: services/shhext/chat/sql_lite_persistence.go [hunk 1]
```diff
@@ -137,7 +149,68 @@ func MigrateDBKeyKdfIterations(oldPath string, newPath string, key string) error
...
-	exportDB := "SELECT sqlcipher_export('newdb')"
+	if _, err = db.Exec(exportDB); err != nil {
+		return err
+	}
+
+	if err = db.Close(); err != nil {
+		return err
+	}
+
+	if err = os.Remove(oldPath); err != nil {
```
Comment (adambabik) [inline services/shhext/chat/sql_lite_persistence.go]: Could be just `return os.Remove(oldPath)`.

File: services/shhext/chat/sql_lite_persistence.go [hunk 2]
```diff
@@ -137,7 +149,68 @@ func MigrateDBKeyKdfIterations(oldPath string, newPath string, key string) error
...
+		return err
+	}
+
+	return nil
+
```
Comment (adambabik) [inline services/shhext/chat/sql_lite_persistence.go]: Empty line.

File: services/shhext/chat/sql_lite_persistence.go [hunk 3]
```diff
@@ -137,7 +149,68 @@ func MigrateDBKeyKdfIterations(oldPath string, newPath string, key string) error
...
+}
+
+// EncryptDatabase encrypts an unencrypted database with key
+func EncryptDatabase(oldPath string, newPath string, key string) error {
+	_, err := os.Stat(oldPath)
+
+	// No files, nothing to do
+	if os.IsNotExist(err) {
+
```
Comment (adambabik) [inline services/shhext/chat/sql_lite_persistence.go]: Empty line.

**Review by adambabik — APPROVED** [review]
Some clean up would be nice but logic looks good 👍

**Pull request merged**